oddt/scoring/descriptors/__init__.py

```diff
@@ -207,6 +209,68 @@ def __reduce__(self):
                                            self.aligned_pairs)
 
 
+class universal_descriptor(object):
+    def __init__(self,
+                 func,
+                 protein=None,
+                 shape=None):
+        """Close contacts descriptor which tallies atoms of type X in certain
+        cutoff from atoms of type Y.
+
+        Parameters
+        ----------
+            func: object
+                A function to be mapped accross all ligands. Can be any callable
+                object.
+
+            protein: oddt.toolkit.Molecule or None (default=None)
+                Default protein to use as reference
+
+        """
+        self.shape = shape
+        self.func = func
+        self.protein = protein
+        if isinstance(func, partial):
+            self.titles = self.func.func.__name__
+        else:
+            self.titles = self.func.__name__
+
+    def build(self, ligands, protein=None):
+        """Builds descriptors for series of ligands
+
+        Parameters
+        ----------
+            ligands: iterable of oddt.toolkit.Molecules or oddt.toolkit.Molecule
+                A list or iterable of ligands to build the descriptor or a
+                single molecule.
+
+            protein: oddt.toolkit.Molecule or None (default=None)
+                Default protein to use as reference
+
+        """
+        if protein:
+            self.protein = protein
+        if is_molecule(ligands):
+            ligands = [ligands]
+        out = []
+        for mol in ligands:
+            if protein is None:
```

Comment:
What if `protein` is None, but `self.protein` isn't?

Reply:
This is fixed now